scene/resources/packed_scene.cpp

```diff
@@ -347,7 +347,27 @@ Node *SceneState::instantiate(GenEditState p_edit_state) const {
 							node->get_script_instance()->get_property_state(old_state);
 						}
 
-						node->set(snames[nprops[j].name], props[nprops[j].value], &valid);
+#ifdef TOOLS_ENABLED
+						const Ref<Script> value_as_script = (Script *)(Object *)props[nprops[j].value];
+						// It is possible that the user changed an existing script to abstract after it was attached to a node.
+						// When this happens, the user needs to fix it, and we should keep track of the global class name to
+						// allow the "Extend Script" button to work. See https://github.com/godotengine/godot/issues/109171
+						if (value_as_script.is_valid() && value_as_script->is_abstract()) {
+							const String global_class_name = value_as_script->get_global_name();
+							if (global_class_name.is_empty()) {
+								ERR_PRINT("Node '" + snames[n.name] + "' previously had a script, but that script is now abstract. Please assign a different script (right-click -> Extend Script...) or change the node to a different type (right-click -> Change Type...) to fix this, then re-save the scene.");
+							} else {
+								ERR_PRINT("Node '" + snames[n.name] + "' previously had a class of type '" + global_class_name + "', but that class is now abstract. Please assign a different script (right-click -> Extend Script...) or change the node to a different type (right-click -> Change Type...) to fix this, then re-save the scene.");
+							}
+							if (node->has_meta(SceneStringName(_custom_type_script))) {
+								node->remove_meta(SceneStringName(_custom_type_script));
```

Comment:
For some reason the meta is not loaded yet here, so it doesn't get removed. You could probably defer it?